Build errors when building in source-build with the latest compiler version.

The following build errors were encountered when building aspnetcore with the latest compiler which is what source-build does.
See https://github.com/dotnet/installer/pull/15151#issuecomment-1371385658 for additional details.

```
  /repos/dotnet/src/aspnetcore/artifacts/source-build/self/src/src/SignalR/common/Shared/ClientResultsManager.cs(26,31): error CS8600: Converting null literal or possible null value to non-nullable type. [/repos/dotnet/src/aspnetcore/artifacts/source-build/self/src/src/SignalR/server/Core/src/Microsoft.AspNetCore.SignalR.Core.csproj]
  /repos/dotnet/src/aspnetcore/artifacts/source-build/self/src/src/SignalR/common/Shared/ClientResultsManager.cs(26,31): error CS8604: Possible null reference argument for parameter 'result' in 'void TaskCompletionSourceWithCancellation<T>.SetResult(T result)'. [/repos/dotnet/src/aspnetcore/artifacts/source-build/self/src/src/SignalR/server/Core/src/Microsoft.AspNetCore.SignalR.Core.csproj]
```

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -159,9 +159,6 @@
     <MicrosoftNETCoreAppRuntimeVersion>$(MicrosoftNETCoreAppRuntimewinx64Version)</MicrosoftNETCoreAppRuntimeVersion>
   </PropertyGroup>
   <PropertyGroup Label="Manual">
-    <!-- Bumping the Roslyn version used in order to ingest the new runtime source generators -->
-    <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
-    <MicrosoftNetCompilersToolsetVersion>4.5.0-1.22517.9</MicrosoftNetCompilersToolsetVersion>
     <!-- DiagnosticAdapter package pinned temporarily (??) until migrated/deprecated -->
     <!-- This is the latest version found in dotnet-public. -->
     <MicrosoftExtensionsDiagnosticAdapterVersion>5.0.0-preview.3.20215.2</MicrosoftExtensionsDiagnosticAdapterVersion>

--- a/src/SignalR/common/Shared/ClientResultsManager.cs
+++ b/src/SignalR/common/Shared/ClientResultsManager.cs
@@ -23,7 +23,7 @@ internal sealed class ClientResultsManager : IInvocationBinder
             var tcs = (TaskCompletionSourceWithCancellation<T>)state;
             if (completionMessage.HasResult)
             {
-                tcs.SetResult((T)completionMessage.Result);
+                tcs.SetResult((T)completionMessage.Result!);
             }
             else
             {

